### DISCUSSION TITLE
Shouldn't ILI9341 SPI write speed be limited to 10MHz?

### DISCUSSION THREAD

**Discussion by KevScott** [Bodmer/TFT_eSPI#1941] [Q&A - General] [answered]
If I look in the User_Setups/SetupX_Template.h file, section 4 says a 40MHz SPI speed should work fine for an ILI9341. 

However, if I look at a datasheet for the device (for example https://cdn-shop.adafruit.com/datasheets/ILI9341.pdf seems to be an up to date version), it says on page 243 that the serial clock cycle period for a write (Twc) should be a minimum of 100ns which suggests a max SPI rate of 10MHz. Maybe there is something I am missing that would allow the 40MHz to be achieved, any clarification would be appreciated?

**Comment by Bodmer** [comment, marked as answer]
You can operate at lower clock rates if you wish. By experiment I found the displays work reliably at 65MHz but I usually run mine at 40MHz.

**Reply by KevScott** [reply]
Thanks for the quick reply. Great library by the way!
